app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,46 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show destroy]
+  before_action :authenticate_user!, except: %i[index show]
+
+  def index
+    @posts = Post.all.order('created_at DESC')
+    @post = Post.new
+  end
+
+  def new
+    @post = current_user.posts.build
```

Comment:
Initializing an empty post with `@post = Post.new` is enough for the `new` action, since `current_user.posts.build` is being used in the `create` action of the post controller.

Reply:
Hello, teekaytech, 
Thanks for the quick review. I can only remove the show from the line of `authenticate_user!` but I can't remove it from routes and `before_action` for a post, It will give me the error because I have used the `show.html.erb` file which uses the link/routes for `post_path(:id)` e.g `<%= link_to post.title, post_path(post) %>`.